src/infra/core/src/ingest/fetch_service/core.rs

```diff
@@ -46,6 +46,48 @@ pub struct FetchService {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+pub(crate) fn template_string<'a>(
+    s: &'a str,
+    lookup_fn: &dyn Fn(&str) -> Result<String, PollingIngestError>,
+) -> Result<Cow<'a, str>, PollingIngestError> {
```

Comment:
I'm not sure PollingIngestError is right error here. LF suggestions.

Reply:
`PollingIngestError` is specific to polling ingest service, fetch service is a part of it,.
Since you are isolating the function, you can define your own error struct, specific for this algorithm. 
Then, it can be converted to whatever where it will be used.